cabal-install/Distribution/Client/ProjectBuilding.hs

```diff
@@ -690,7 +709,8 @@ rebuildTargets verbosity
     packageDBsToUse = -- all the package dbs we may need to create
       (Set.toList . Set.fromList)
         [ pkgdb
-        | InstallPlan.Configured pkg <- InstallPlan.toList installPlan
+        | InstallPlan.Configured pkg_or_comp <- InstallPlan.toList installPlan
+        , let pkg = getElaboratedPackage pkg_or_comp
```

Comment:
Lots of examples like this. If we could better factor the package vs component types of elaborated package then we wouldn't have this kind of noise everywhere.